The color logger might not be a great idea for Shippable

```
2019-01-15 15:51:04.712225 | static | _______________ TestInventoryUpdateCredentials.test_awx_task_env _______________
2019-01-15 15:51:04.712313 | static | [gw3] linux2 -- Python 2.7.5 /venv/awx/bin/python2
2019-01-15 15:51:04.712347 | static | Traceback (most recent call last):
2019-01-15 15:51:04.712409 | static |   File "/awx_devel/awx/main/tests/unit/test_tasks.py", line 2264, in test_awx_task_env
2019-01-15 15:51:04.712436 | static |     self.task.run(self.pk)
2019-01-15 15:51:04.712482 | static |   File "/awx_devel/awx/main/tasks.py", line 619, in _wrapped
2019-01-15 15:51:04.712512 | static |     return f(self, *args, **kwargs)
2019-01-15 15:51:04.712579 | static |   File "/awx_devel/awx/main/tasks.py", line 1073, in run
2019-01-15 15:51:04.712617 | static |     raise AwxTaskError.TaskError(instance, rc)
2019-01-15 15:51:04.712691 | static | Exception: inventory_update None (error) encountered an error (rc=None), please see task stdout for details.
2019-01-15 15:51:04.712811 | static | ----------------------------- Captured stderr call -----------------------------
2019-01-15 15:51:04.712910 | static | [31;1m2019-01-15 15:47:58,937 ERROR    awx.main.tasks inventory_update None (running) Exception occurred while running task[0m
2019-01-15 15:51:04.712947 | static | [31;1mTraceback (most recent call last):[0m
2019-01-15 15:51:04.712996 | static | [31;1m  File "/awx_devel/awx/main/tasks.py", line 964, in run[0m
2019-01-15 15:51:04.713101 | static | [31;1m    raise Exception('alan!')  # I will delete this, but I need to verify test output first[0m
2019-01-15 15:51:04.713129 | static | [31;1mException: alan![0m
2019-01-15 15:51:04.713214 | static | [1m2019-01-15 15:47:58,938 INFO     awx.main.tasks inventory_update None (running) finished running, producing 0 events.[0m
```

But it is useful locally... I wouldn't mind keeping it this way personally.

--- a/awx/main/models/inventory.py
+++ b/awx/main/models/inventory.py
@@ -3,6 +3,7 @@
 
 # Python
 import datetime
+import time
 import logging
 import re
 import copy
@@ -428,7 +429,8 @@ class Inventory(CommonModelNameNotUnique, ResourceMixin, RelatedJobsMixin):
         '''
         Update model fields that are computed from database relationships.
         '''
-        logger.debug("Going to update inventory computed fields")
+        logger.debug("Going to update inventory computed fields, pk={0}".format(self.pk))
+        start_time = time.time()
         if update_hosts:
             self.update_host_computed_fields()
         if update_groups:
@@ -465,7 +467,8 @@ class Inventory(CommonModelNameNotUnique, ResourceMixin, RelatedJobsMixin):
                 computed_fields.pop(field)
         if computed_fields:
             iobj.save(update_fields=computed_fields.keys())
-        logger.debug("Finished updating inventory computed fields")
+        logger.debug("Finished updating inventory computed fields, pk={0}, in "
+                     "{1:.3f} seconds".format(self.pk, time.time() - start_time))
 
     def websocket_emit_status(self, status):
         connection.on_commit(lambda: emit_channel_notification(

--- a/awx/main/tests/unit/test_tasks.py
+++ b/awx/main/tests/unit/test_tasks.py
@@ -246,8 +246,6 @@ class TestJobExecution(object):
             # If `Job.update_model` is called, we're not actually persisting
             # to the database; just update the status, which is usually
             # the update we care about for testing purposes
-            if kwargs.get('result_traceback'):
-                raise Exception('Task encountered error:\n{}'.format(kwargs['result_traceback']))
             if 'status' in kwargs:
                 self.instance.status = kwargs['status']
             if 'job_env' in kwargs:

--- a/awx/main/utils/common.py
+++ b/awx/main/utils/common.py
@@ -10,7 +10,6 @@ import os
 import re
 import subprocess
 import stat
-import sys
 import urllib
 import urlparse
 import threading
@@ -118,7 +117,7 @@ class RequireDebugTrueOrTest(logging.Filter):
 
     def filter(self, record):
         from django.conf import settings
-        return settings.DEBUG or 'test' in sys.argv
+        return settings.DEBUG or settings.IS_TESTING()
 
 
 class IllegalArgumentError(ValueError):

--- a/awx/settings/development.py
+++ b/awx/settings/development.py
@@ -45,6 +45,8 @@ SHELL_PLUS_PRINT_SQL = False
 # show colored logs in the dev environment
 # to disable this, set `COLOR_LOGS = False` in awx/settings/local_settings.py
 LOGGING['handlers']['console']['()'] = 'awx.main.utils.handlers.ColorHandler'
+# task system does not propagate to AWX, so color log these too
+LOGGING['handlers']['task_system'] = LOGGING['handlers']['console'].copy()
 COLOR_LOGS = True
 
 # Pipe management playbook output to console

--- a/awx/settings/local_settings.py.docker_compose
+++ b/awx/settings/local_settings.py.docker_compose
@@ -91,9 +91,6 @@ CHANNEL_LAYERS = {
                 'CONFIG': {'url': BROKER_URL}}
 }
 
-# Set True to enable additional logging from the job_event_callback plugin
-JOB_CALLBACK_DEBUG = False
-
 # Absolute filesystem path to the directory to host projects (with playbooks).
 # This directory should NOT be web-accessible.
 PROJECTS_ROOT = '/projects/'
@@ -143,9 +140,6 @@ PROXY_IP_WHITELIST = []
 
 # If set, use -vvv for project updates instead of -v for more output.
 # PROJECT_UPDATE_VVV=True
-
-# Set verbosity for inventory import command when running inventory updates.
-# INVENTORY_UPDATE_VERBOSITY=1
 
 ###############################################################################
 # EMAIL SETTINGS
@@ -221,9 +215,9 @@ LOGGING['handlers']['management_playbooks'] = {'class': 'logging.NullHandler'}
 #}
 
 # Enable the following lines to turn on lots of permissions-related logging.
-#LOGGING['loggers']['awx.main.access']['propagate'] = True
-#LOGGING['loggers']['awx.main.signals']['propagate'] = True
-#LOGGING['loggers']['awx.main.permissions']['propagate'] = True
+#LOGGING['loggers']['awx.main.access']['level'] = 'DEBUG'
+#LOGGING['loggers']['awx.main.signals']['level'] = 'DEBUG'
+#LOGGING['loggers']['awx.main.permissions']['level'] = 'DEBUG'
 
 # Enable the following line to turn on database settings logging.
 #LOGGING['loggers']['awx.conf']['level'] = 'DEBUG'

--- a/awx/settings/local_settings.py.example
+++ b/awx/settings/local_settings.py.example
@@ -48,9 +48,6 @@ if is_testing(sys.argv):
 # AMQP configuration.
 BROKER_URL = 'amqp://guest:guest@localhost:5672'
 
-# Set True to enable additional logging from the job_event_callback plugin
-JOB_CALLBACK_DEBUG = False
-
 # Absolute filesystem path to the directory to host projects (with playbooks).
 # This directory should NOT be web-accessible.
 PROJECTS_ROOT = os.path.join(BASE_DIR, 'projects')
@@ -99,9 +96,6 @@ PROXY_IP_WHITELIST = []
 
 # If set, use -vvv for project updates instead of -v for more output.
 # PROJECT_UPDATE_VVV=True
-
-# Set verbosity for inventory import command when running inventory updates.
-# INVENTORY_UPDATE_VERBOSITY=1
 
 ###############################################################################
 # EMAIL SETTINGS
@@ -158,9 +152,9 @@ LOGGING['handlers']['syslog'] = {
 #}
 
 # Enable the following lines to turn on lots of permissions-related logging.
-#LOGGING['loggers']['awx.main.access']['propagate'] = True
-#LOGGING['loggers']['awx.main.signals']['propagate'] = True
-#LOGGING['loggers']['awx.main.permissions']['propagate'] = True
+#LOGGING['loggers']['awx.main.access']['level'] = 'DEBUG'
+#LOGGING['loggers']['awx.main.signals']['level'] = 'DEBUG'
+#LOGGING['loggers']['awx.main.permissions']['level'] = 'DEBUG'
 
 # Enable the following line to turn on database settings logging.
 #LOGGING['loggers']['awx.conf']['level'] = 'DEBUG'

